oshdb/src/test/java/org/heigit/ohsome/oshdb/contribution/ContributionsRelationTest.java

```diff
@@ -0,0 +1,315 @@
+package org.heigit.ohsome.oshdb.contribution;
+
+import static java.util.Collections.emptyList;
+import static org.heigit.ohsome.oshdb.contribution.ContributionType.CREATION;
+import static org.heigit.ohsome.oshdb.contribution.ContributionType.DELETION;
+import static org.heigit.ohsome.oshdb.contribution.ContributionType.GEOMETRY_CHANGE;
+import static org.heigit.ohsome.oshdb.contribution.ContributionType.MEMBER_CHANGE;
+import static org.heigit.ohsome.oshdb.contribution.ContributionType.ROLE_CHANGE;
+import static org.heigit.ohsome.oshdb.contribution.ContributionType.TAG_CHANGE;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.EnumSet;
+import java.util.List;
+import org.heigit.ohsome.oshdb.OSHDBTest;
+import org.junit.jupiter.api.Test;
+
+class ContributionsRelationTest extends OSHDBTest {
+
+  @Test
+  void testMemberTagChanges() {
+    var nodes = List.of(
+        osh(1,
+            node(1, 1000, 101, 1, tags(), 0, 0)));
+    var ways = List.of(
+        osh(1, List.of(
+            osh(1,
+                node(3, 3500, 253, 3, tags(), 1, 1),
+                node(2, 2500, 253, 3, tags(), 1, 1), // squash
+                node(1, 1100, 101, 1, tags(), 0, 0))), // minor-major change
+            way(1, 1000, 101, 1, tags(), mems(1))));
+
+    var versions = relations(1,
+        relation(2, 2000, 202, 2, tags(tag(1, 1)), mems(w(1, 0))),
+        relation(1, 1000, 101, 1, tags(), mems(w(1, 0), n(1, 0))));
+    var osh = osh(versions, nodes, ways);
+    var contribs = Contributions.of(osh);
+
+    assertTrue(contribs.hasNext());
+    var contrib = contribs.next();
+    assertContrib(3500, 253, 3, versions.get(0), contrib);
```

Comment:
> wouldn't it be more precise to say the geometry is already changed at time 2500 and we would squash the changes into the older timestamp? What would be the downside of that?

It is unfortunately not so clear in this test, you are right, but consider that at timestamp 3500 there is a geometry change as well
```
osh(1,
  node(3, 3500, 253, 3, tags(), 2, 2),
  node(2, 2500, 253, 3, tags(), 1, 1), // squash
  node(1, 1100, 101, 1, tags(), 0, 0)))
```
I think it is worth for a discussion and decision for squashing in general!
In the current implementation you miss the geometry (1,1) which, in my opinion this is ok, even it would have been visible at the map for a short time, but as the intention of this change (253) was to change the geometry from (0,0) to (2,2)  and the change finished at 3500.